src/H5MF.c

```diff
@@ -1057,7 +1057,9 @@ H5MF_xfree(H5F_t *f, H5FD_mem_t alloc_type, haddr_t addr, hsize_t size)
     assert(f);
     if (!H5_addr_defined(addr) || 0 == size)
         HGOTO_DONE(SUCCEED);
-    assert(addr != 0); /* Can't deallocate the superblock :-) */
+
+    if (addr <= 0)
+        HGOTO_ERROR(H5E_RESOURCE, H5E_BADRANGE, FAIL, "attempting to free file superblock");
```

Comment:
I know the original assert didn't have, but should addr also be checked for upper bound, not going pass the end of the file?